### ISSUE TITLE
MagnitudePruner/MagnitudeImportance vs GroupNormPruner/GroupNormImportance

### ISSUE THREAD

**Issue by Serjio42** [VainF/Torch-Pruning#117]
Hi, @VainF . Nice to see the updated library version and the paper preprint!

I'm trying to figure out the difference between code of `MagnitudePruner` and `GroupNormPruner` implemented in the current repo code.
To be honest, after code revising of both I did not get the core difference between them: each looks like analyzing **the group channels as a whole** and prunes the least important channels in the group.
I've made an experiment to compare both strategies on pretrained resnet and the results look the same. Could you please clarify the case? Here is the code of my experiment:
```
import torch
from torchvision.models import resnet18
import torch_pruning as tp

def compare_models(model_1, model_2):
    models_differ = 0
    for key_item_1, key_item_2 in zip(model_1.state_dict().items(), model_2.state_dict().items()):
        if torch.equal(key_item_1[1], key_item_2[1]):
            pass
        else:
            models_differ += 1
            if (key_item_1[0] == key_item_2[0]):
                print('Mismtach found at', key_item_1[0])
            else:
                raise Exception
    if models_differ == 0:
        print('Models match perfectly! :)')
        
example_inputs = torch.randn(1,3,224,224)

# THE FIRST CASE
model = resnet18(weights="IMAGENET1K_V1").eval()
imp = tp.importance.MagnitudeImportance(p=2, group_reduction='mean')
ignored_layers = []
iterative_steps = 1
for i in range(iterative_steps):
    pruner = tp.pruner.MagnitudePruner(
        model, 
        example_inputs, 
        global_pruning=False, # If False, a uniform sparsity will be assigned to different layers.
        importance=imp, # importance criterion for parameter selection
        iterative_steps=1, # the number of iterations to achieve target sparsity
        ch_sparsity=0.05,
        ignored_layers=ignored_layers,
        )
    pruner.step()

# THE SECOND CASE
model_gr = resnet18(weights="IMAGENET1K_V1").eval()
imp_gr = tp.importance.GroupNormImportance(p=2)
ignored_layers = []
iterative_steps = 1
for i in range(iterative_steps):
    pruner = tp.pruner.GroupNormPruner(
        model_gr, 
        example_inputs, 
        global_pruning=False, # If False, a uniform sparsity will be assigned to different layers.
        importance=imp_gr, # importance criterion for parameter selection
        iterative_steps=1, # the number of iterations to achieve target sparsity
        ch_sparsity=0.05,
        ignored_layers=ignored_layers,
        )
    pruner.step()

compare_models(model, model_gr)
```

**Comment by VainF** [comment]
BTW, I think it would be more reasonable to get an unreduced importance matrix from MagnitudeImportance(group_reduction=None). I will try to modify it! Haha.

**Comment by Serjio42** [comment]
@VainF thanks for providing additional info.
The one thing I still didn't get is this "**sparse training**" term, is it when you just add L1-regularization to zero out unimportant channels (like in [this work](https://arxiv.org/pdf/1708.06519.pdf) with BatchNorm scale factor utilizing)? If not, where I can read/learn what it means?
